File: backend/app/schemas/website.py
"""Pydantic schemas for website intake (public.quotation_requests)."""
import uuid
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict, Field

QuotationRequestStatus = Literal[
    "new",
    "contacted",
    "quotation_preparing",
    "quotation_sent",
    "won",
    "lost",
    "closed",
]


class QuotationRequestOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    reference: str
    full_name: str
    organization: str
    phone: str
    email: str | None
    organization_type: str | None
    requirement: str
    consent: bool
    consent_at: datetime
    source_page: str | None
    product_slug: str | None
    category_slug: str | None
    product_name: str | None
    intent: str | None
    status: str
    internal_notes: str | None
    contacted_at: datetime | None
    created_at: datetime
    updated_at: datetime


class QuotationRequestUpdate(BaseModel):
    """The only columns the CRM may write. Everything else is the website's."""

    status: QuotationRequestStatus | None = None
    internal_notes: str | None = Field(default=None, max_length=5000)


class QuotationRequestSummary(BaseModel):
    total: int
    # status -> count, every known status present (zero when unused).
    by_status: dict[str, int]
    new_count: int
    open_count: int

File: frontend/src/lib/types/website.ts
/** Website intake — rows of public.quotation_requests, served by FastAPI. */

export type QuotationRequestStatus =
  | "new"
  | "contacted"
  | "quotation_preparing"
  | "quotation_sent"
  | "won"
  | "lost"
  | "closed";

export const QUOTATION_REQUEST_STATUSES: QuotationRequestStatus[] = [
  "new",
  "contacted",
  "quotation_preparing",
  "quotation_sent",
  "won",
  "lost",
  "closed",
];

export const QUOTATION_REQUEST_STATUS_LABELS: Record<QuotationRequestStatus, string> = {
  new: "New",
  contacted: "Contacted",
  quotation_preparing: "Preparing quotation",
  quotation_sent: "Quotation sent",
  won: "Won",
  lost: "Lost",
  closed: "Closed",
};

/** Only the website writes these; the CRM writes status/internal_notes/contacted_at. */
export interface QuotationRequest {
  id: string;
  reference: string;
  full_name: string;
  organization: string;
  phone: string;
  email: string | null;
  organization_type: string | null;
  requirement: string;
  consent: boolean;
  consent_at: string;
  source_page: string | null;
  product_slug: string | null;
  category_slug: string | null;
  product_name: string | null;
  intent: string | null;
  status: QuotationRequestStatus;
  internal_notes: string | null;
  contacted_at: string | null;
  created_at: string;
  updated_at: string;
}

export interface QuotationRequestSummary {
  total: number;
  by_status: Record<QuotationRequestStatus, number>;
  new_count: number;
  open_count: number;
}
